### DISCUSSION TITLE
Enhance UserModel to Include Roles and Permissions

### DISCUSSION THREAD

**Discussion by PlaxXOnline** [serverpod/serverpod#2565] [Q&A] [answered]
Is your feature request related to a problem? Please describe.
I’m always frustrated when trying to implement a role or permission system in Serverpod. Currently, it’s only possible by creating an additional table to extend the UserModel. This adds complexity and makes the implementation of a robust role or permission system more cumbersome than necessary.

Describe the solution you’d like
I would like to see an extension of the existing UserModel to include roles and permissions directly within it. This would simplify the process of implementing a role or permission system, making it more straightforward and integrated.

Describe alternatives you’ve considered
An alternative solution I’ve considered is maintaining a separate table for roles and permissions, but this approach increases the complexity of the database schema and the overall implementation. It also requires additional coding to manage the relationships and ensure data integrity.

Additional context
Incorporating roles and permissions directly into the UserModel would streamline the development process and improve maintainability. It would also align with best practices seen in other frameworks and systems, making Serverpod more competitive and user-friendly.

**Comment by Isakdl** [comment, marked as answer]
Can you expand a little on what you mean here?

The UserInfo model does have the scopes in the model already! and you can easily modify them like documented here: https://docs.serverpod.dev/concepts/authentication/basics#managing-scopes

The field is called `scopeNames` on the `UserInfo` model.

Is this not enough for your use case?

**Reply by PlaxXOnline** [reply]
Thank you for your advise!

The Scopes are exactly what i´m searching for. :)
